src/runtime_src/core/pcie/driver/linux/xocl/subdev/firewall.c

```diff
@@ -229,65 +221,36 @@ static int get_prop(struct platform_device *pdev, u32 prop, void *val)
 	fw = platform_get_drvdata(pdev);
 	BUG_ON(!fw);
 
-	if (FW_PRIVILEGED(fw)) {
-
-		(void) check_firewall(pdev, NULL);
-
-		switch (prop) {
-		case XOCL_AF_PROP_TOTAL_LEVEL:
-			*(u64 *)val = fw->max_level;
-			break;
-		case XOCL_AF_PROP_STATUS:
-			*(u64 *)val = fw->curr_status;
-			break;
-		case XOCL_AF_PROP_LEVEL:
-			*(int64_t *)val = fw->curr_level;
-			break;
-		case XOCL_AF_PROP_DETECTED_STATUS:
-			*(u64 *)val = fw->err_detected_status;
-			break;
-		case XOCL_AF_PROP_DETECTED_LEVEL:
-			*(u64 *)val = fw->err_detected_level;
-			break;
-		case XOCL_AF_PROP_DETECTED_TIME:
-			*(u64 *)val = fw->err_detected_time;
-			break;
-		case XOCL_AF_PROP_DETECTED_LEVEL_NAME:
-			strcpy((char *)val, fw->err_detected_level_name);
-			break;
-		default:
-			xocl_err(&pdev->dev, "Invalid prop %d", prop);
-			ret = -EINVAL;
-		}
-	} else {
-		get_fw_status(pdev);
+	/* If not privileged request the firewall status from the mgmt driver */
+	if (!FW_PRIVILEGED(fw))
+		request_firewall_status(pdev);
```

Comment:
This should be removed. We do not need to request firewall info for sysfs accessing